x-pack/filebeat/input/azureblobstorage/job.go

```diff
@@ -72,6 +87,9 @@ func (j *job) do(ctx context.Context, id string) {
 		if *j.blob.Properties.ContentType == gzType || (j.blob.Properties.ContentEncoding != nil && *j.blob.Properties.ContentEncoding == encodingGzip) {
 			j.isCompressed = true
 		}
+		isRootArray, done := j.state.isRootArray(*j.blob.Name)
+		done()
```

Comment:
Is this safe to release prior to the `processAndPublishData` call? I think it has to be or we have a deadlock when we get to `readJsonAndPublish`. Can you check this?

Reply:
yes, it causes a deadlock if we don't release. And yea it is safe to release it here since we do not perform any op on this this map until the job is complete and the entry is removed, which is again protected with locks.